aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -514,4 +530,22 @@ public void setConf(Configuration conf) {
   protected Map<String, String> properties() {
     return catalogProperties == null ? ImmutableMap.of() : catalogProperties;
   }
+
+  private boolean isTablePurged(TableIdentifier identifier, S3Client s3Client) {
+    Table table = glue.getTable(GetTableRequest.builder()
+        .catalogId(awsProperties.glueCatalogId())
+        .databaseName(identifier.namespace().toString())
+        .name(identifier.name())
+        .build()).table();
+    String metadataLocation = table.parameters().get(BaseMetastoreTableOperations.METADATA_LOCATION_PROP);
+
+    try {
+      S3URI s3URI = new S3URI(metadataLocation);
```

Comment:
we don't need to make `S3URI` and `S3FileIO.client()` public, we can just do `fileIO.newInputFile(...).exists()`